woocommerce/woocommerce
# Add include_meta and exclude_meta params for orders and products

### Changes proposed in this Pull Request:

Introduces two new collection params for v2 and v3 orders and products REST API endpoints. These params allow for limiting which meta keys are included in the `meta_data` property of the API responses.

Fixes #34243

### How to test the changes in this Pull Request:

1. Create some products and orders in your test environment.
2. On at least one order and at least one product, add at least two postmeta key/value pairs using the Custom Fields box. E.g. `test1` and `test2`.
3. Using a REST client like Insomnia, make a request to `wc/v3/orders` and check that the postmeta keys you added are showing up under the `meta_data` property of the order in the response. Do the same for `wc/v3/products`. It's easier to find the data you're looking for if you use the `_fields` request param to limit which fields are in the response to something like `id,meta_data`.
4. Add the `include_meta` param to your requests with a value that is only one of the postmeta keys you added, e.g. `test1`. Now the response should only include `test1` in the `meta_data` prop.
5. Switch the param to `exclude_meta`. Now the response should only include `test2` in the `meta_data` prop.
6. Try using both `include_meta` and `exclude_meta` params in the request. The `include` one should be honored, while the `exclude` one is ignored.

### Other information:


### FOR PR REVIEWER ONLY:
